src/huggingface_hub/serialization/_torch.py

```diff
@@ -335,18 +335,22 @@ def split_torch_state_dict_into_shards(
         get_storage_id=get_torch_storage_id,
     )
 
-
-def get_torch_storage_id(tensor: "torch.Tensor") -> Tuple["torch.device", int, int]:
+def _get_unique_id(tensor: "torch.Tensor") -> Union[int, Tuple[Any, ...]]:
+    """Returns a unique id for plain tensor
+    or a (potentially nested) Tuple of unique id for the flattened Tensor
+    if the input is a wrapper tensor subclass Tensor
     """
-    Return unique identifier to a tensor storage.
 
-    Multiple different tensors can share the same underlying storage. For
-    example, "meta" tensors all share the same storage, and thus their identifier will all be equal. This identifier is
-    guaranteed to be unique and constant for this tensor's storage during its lifetime. Two tensor storages with
-    non-overlapping lifetimes may have the same id.
+    try:
+        # for torch 2.1 and above we can also handle tensor subclasses
+        from torch.utils._python_dispatch import is_traceable_wrapper_subclass
+        if is_traceable_wrapper_subclass(tensor):
+            attrs, _ = tensor.__tensor_flatten__()
+            unique_id = tuple(_get_unique_id(getattr(tensor, attr)) for attr in attrs)
```

Comment:
Shouldn't we return directly here? When I read the implementation [here](https://github.com/huggingface/huggingface_hub/blob/ca3c22801e964ec831d0361c08c50cecd1bd6522/src/huggingface_hub/serialization/_torch.py#L349), `unique_id` seems to be overwritten in all cases

Reply:
oh sorry, yeah I meant to return here